### PR TITLE
Added a notification system for when shifts are published

### PR DESCRIPTION
**Pull request by Christopher-Chianelli**

### CODE REVIEW

**Comment by manstis** [comment]
We have two other Notification systems in kie-wb-common; it might be nice to consider reuse of one. Is there good reason not to?

**Comment by Christopher-Chianelli** [comment]
@manstis Agreed, but I dunno if I can port the notification system time-wise

**Comment by Christopher-Chianelli** [comment]
(for this to be ready for Summit)

File: optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/common/JQuery.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package org.optaplanner.openshift.employeerostering.gwtui.client.common;
+
+import elemental2.dom.HTMLElement;
+import jsinterop.annotations.JsMethod;
+import jsinterop.annotations.JsPackage;
+import jsinterop.annotations.JsType;
+
+@JsType(isNative = true, namespace = JsPackage.GLOBAL)
+public class JQuery {
```
Comment (ge0ffrey) [inline optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/common/JQuery.java]: I would expect that Errai might have support for this already? @mbarkley

File: optashift-employee-rostering-server/src/main/java/org/optaplanner/openshift/employeerostering/server/roster/RosterRestServiceImpl.java [hunk 2]
```diff
@@ -339,12 +340,14 @@ public void updateShiftsOfRoster(Roster newRoster) {
 
     @Override
     @Transactional
-    public void publishAndProvision(Integer tenantId) {
+    public List<LocalDate> publishAndProvision(Integer tenantId) {
```
Comment (ge0ffrey) [inline optashift-employee-rostering-server/src/main/java/org/optaplanner/openshift/employeerostering/server/roster/RosterRestServiceImpl.java]: Let's use a strongly typed approach. Here we are hardcoding that that is a List of 2 values, and the elements are not self-explenatory.

Let's return a new class PublishResult with has 2 LocalDate fields.

Comment (Christopher-Chianelli) [reply optashift-employee-rostering-server/src/main/java/org/optaplanner/openshift/employeerostering/server/roster/RosterRestServiceImpl.java]: Done

File: optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/common/JQuery.java (hunk 1 above)
Comment (mbarkley) [reply optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/common/JQuery.java]: We have an example like this in the Errai Tutorial, but we don't provide wrappers for any JavaScript libraries in Errai.

File: optashift-employee-rostering-shared/src/main/java/org/optaplanner/openshift/employeerostering/shared/roster/PublishResult.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
+package org.optaplanner.openshift.employeerostering.shared.roster;
+
+import java.time.LocalDate;
+
+public class PublishResult {
+    private LocalDate publishedFromInclusive;
```
Comment (ge0ffrey) [inline optashift-employee-rostering-shared/src/main/java/org/optaplanner/openshift/employeerostering/shared/roster/PublishResult.java]: => publishedFromDate

File: optashift-employee-rostering-shared/src/main/java/org/optaplanner/openshift/employeerostering/shared/roster/PublishResult.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    private LocalDate publishedToExclusive;
```
Comment (ge0ffrey) [inline optashift-employee-rostering-shared/src/main/java/org/optaplanner/openshift/employeerostering/shared/roster/PublishResult.java]: => publishedToDate // exclusive

File: optashift-employee-rostering-shared/src/main/java/org/optaplanner/openshift/employeerostering/shared/roster/PublishResult.java (hunk 3 above)
Comment (mbarkley) [reply optashift-employee-rostering-shared/src/main/java/org/optaplanner/openshift/employeerostering/shared/roster/PublishResult.java]: Perhaps, `inclusivePublishedStart` (and `exclusivePublishEnd`)? I don't think it hurts to have the inclusive/exclusive in the name.

Comment (Christopher-Chianelli) [reply optashift-employee-rostering-shared/src/main/java/org/optaplanner/openshift/employeerostering/shared/roster/PublishResult.java]: It to be consistent with Shift/EmployeeAvaliability/All other date intervals fields; The inclusiveness of start and the exclusiveness of end is implict; if we were to change it here, we need to change it everywhere

**Pull request merged**